airbyte-integrations/connectors/source-airtable/source_airtable/spec.json

```diff
@@ -4,15 +4,69 @@
     "$schema": "http://json-schema.org/draft-07/schema#",
     "title": "Airtable Source Spec",
     "type": "object",
-    "required": ["api_key"],
+    "additionalProperties": true,
     "properties": {
-      "api_key": {
-        "type": "string",
-        "description": "The API Key or PAT for the Airtable account. See the <a href=\"https://airtable.com/developers/web/guides/personal-access-tokens\">Support Guide</a> for more information on how to obtain this key.",
-        "title": "API Key",
-        "airbyte_secret": true,
-        "examples": ["key1234567890"]
+      "credentials": {
+        "title": "Authentication",
+        "type": "object",
+        "oneOf": [
+          {
+            "type": "object",
+            "title": "OAuth2.0",
+            "required": ["client_id", "client_secret", "refresh_token"],
+            "properties": {
+              "auth_method": {
+                "type": "string",
+                "const": "oauth2.0"
+              },
+              "client_id": {
+                "type": "string",
+                "title": "Client ID",
+                "description": "The client ID of the LinkedIn Ads developer application.",
+                "airbyte_secret": true
+              },
+              "client_secret": {
+                "type": "string",
+                "title": "Client secret",
+                "description": "The client secret the LinkedIn Ads developer application.",
+                "airbyte_secret": true
+              },
+              "refresh_token": {
+                "type": "string",
+                "title": "Refresh token",
+                "description": "The key to refresh the expired access token.",
+                "airbyte_secret": true
+              }
+            }
+          },
+          {
+            "title": "API Key",
+            "type": "object",
+            "required": ["api_key"],
+            "properties": {
+              "auth_method": {
+                "type": "string",
+                "const": "api_key"
+              },
+              "api_key": {
```

Comment:
if we know its permanent length or some RegExp, we can set this additional validation here.

Reply:
No need to RegExp of alphanumeric chars, especially if this could be changed on the API side, this would require the spec tunings later on, which is not necessary I think.